core/src/crypto.rs

```diff
@@ -25,8 +25,12 @@ impl Secret {
         self.secret.diffie_hellman(pub_key)
     }
 
-    pub fn gen_cipher(&mut self) -> Aes256 {
+    pub fn gen_pub_cipher(&mut self) -> Aes256 {
```

Comment:
Perhaps create a generic `gen_cipher` method that both these methods utilize? Just a thought, but makes for more modularized code and easier maintenance.